File: richpath/src/main/java/com/richpath/pathparser/PathParserCompatApi21.java
package com.richpath.pathparser;

import android.graphics.Path;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

/**
 * Created by tarek on 6/27/17.
 */

class PathParserCompatApi21 {

    /**
     * @param pathData The string representing a path, the same as "d" string in svg file.
     * @return the generated Path object.
     */
    static Path createPathFromPathData(String pathData) {

        try {
            Method method = getCreatePathFromPathDataMethod();
            if (method != null) {
                Object object = method.invoke(null, pathData);
                return (Path) object;
            }
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
        return null;
    }

    private static Method getCreatePathFromPathDataMethod() {
        try {
            return Class.forName("android.util.PathParser")
                    .getDeclaredMethod("createPathFromPathData", String.class);
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        } catch (NoSuchMethodException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: richpath/src/main/kotlin/richpath/pathparser/PathParserCompatApi21.kt
package richpath.pathparser

import android.graphics.Path
import java.lang.reflect.InvocationTargetException
import java.lang.reflect.Method

object PathParserCompatApi21 {

    /**
     * @param pathData The string representing a path, the same as "d" string in svg file.
     * @return the generated Path object.
     */
    fun createPathFromPathData(pathData: String?): Path? {
        try {
            val method = getCreatePathFromPathDataMethod() ?: return null
            val obj = method.invoke(null, pathData)
            return obj as? Path
        } catch (e: InvocationTargetException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
        return null
    }

    private fun getCreatePathFromPathDataMethod(): Method? {
        try {
            return Class.forName("android.util.PathParser")
                    .getDeclaredMethod("createPathFromPathData", String::class.java)
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        } catch (e: NoSuchMethodException) {
            e.printStackTrace()
        }
        return null
    }
}